packages/components/src/components/multi-select/multi-select-option/multi-select-option.tsx

```diff
@@ -0,0 +1,66 @@
+import { Component, Element, h, type JSX, Prop } from '@stencil/core';
+import { AllowedTypes, attachComponentCss, getPrefixedTagNames, validateProps } from '../../../utils';
+import { MultiSelectOptionInternalHTMLProps } from './multi-select-option-utils';
+import type { PropTypes } from '../../../types';
+import { ValidatorFunction } from '../../../types';
+import { getComponentCss } from './multi-select-option-styles';
+import { getOptionAriaAttributes } from '../../../utils/a11y/select/select-aria';
+
+const propTypes: PropTypes<typeof MultiSelectOption> = {
+  value: AllowedTypes.oneOf<ValidatorFunction>([AllowedTypes.string, AllowedTypes.number]),
+  disabled: AllowedTypes.boolean,
+};
+
+@Component({
+  tag: 'p-multi-select-option',
+  shadow: true,
+})
+export class MultiSelectOption {
+  @Element() public host!: HTMLElement & MultiSelectOptionInternalHTMLProps;
+
+  /** The option value. */
+  @Prop() public value: string | number;
+
+  /** Disables the option. */
+  @Prop() public disabled?: boolean = false; // eslint-disable-line @typescript-eslint/no-inferrable-types
+
+  public render(): JSX.Element {
+    validateProps(this, propTypes);
+    attachComponentCss(this.host, getComponentCss, this.host.theme || 'light');
+
+    const PrefixedTagNames = getPrefixedTagNames(this.host);
+
+    return (
+      <div
+        class={{
+          option: true,
+          'option--selected': this.host.selected,
+          'option--highlighted': this.host.highlighted,
+          'option--disabled': this.disabled,
+        }}
+        onClick={!this.disabled && this.onClick}
+      >
+        <PrefixedTagNames.pCheckboxWrapper class="checkbox" theme={this.host.theme || 'light'}>
+          <input
+            role="option"
+            type="checkbox"
+            checked={this.host.selected}
+            disabled={this.disabled}
+            {...getOptionAriaAttributes(this.host.selected, this.disabled, this.host.hidden, !!this.value)}
+            aria-label={this.host.textContent}
+            tabindex="-1"
+          />
+          <slot slot="label" />
+        </PrefixedTagNames.pCheckboxWrapper>
+      </div>
+    );
+  }
+
+  private onClick = (): void => {
+    this.host.dispatchEvent(
+      new CustomEvent('internalOptionUpdate', {
```

Comment:
Let's extract the event name into a reusable const.

Reply:
Stencil Listen decorator currently does not work with constants ([https://github.com/ionic-team/stencil/issues/2924](https://github.com/ionic-team/stencil/issues/2924))